### PR TITLE
Adding label_map to dataset

### LINKED ISSUE #871: [FEAT][CV] Add option to convert class_id to class names
Add a VisionData property named something like "class_label_dict" that will enable checks to translate from class_id (returned by the dataloader as label and by the model predictions) to class names for use in displays. 

Adapt check displays to use this dict if supplied.

### PR DESCRIPTION
**Pull request by ItayGabbay**
#### Reference Issues/PRs

Closes #871 

#### What does this implement/fix? Explain your changes.
Adding mapping from label_id to label_name

#### Any other comments?

---
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/deepchecks/deepchecks/blob/main/CONTRIBUTING.rst

### CODE REVIEW

File: deepchecks/vision/datasets/detection/coco.py [hunk 1]
```diff
@@ -293,3 +294,190 @@ def yolo_image_formatter(pil_list):
     """Convert list of PIL images to deepchecks image format."""
     # Yolo works on PIL and VisionDataset expects images as numpy arrays
     return [np.array(x) for x in pil_list]
+
+
+LABEL_MAP = {
+    0: 'unlabeled',
```
Comment (noamzbr) [inline deepchecks/vision/datasets/detection/coco.py:300]: I thought for some reason this COCO had 80 classes?

File: docs/source/examples/vision/checks/distribution/train_test_label_drift.ipynb [hunk 2]
```diff
@@ -273,16 +273,16 @@
          "name": "Train Dataset",
```
Comment (noamzbr) [inline docs/source/examples/vision/checks/distribution/train_test_label_drift.ipynb:273]: <p>This display (num per class) can't handle this number of classes. Need to add to it the same logic we have in metrics...</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/877/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

Comment (ItayGabbay) [reply docs/source/examples/vision/checks/distribution/train_test_label_drift.ipynb:273]: Let's open a different issue for that

File: deepchecks/vision/dataset.py [hunk 3]
```diff
@@ -187,6 +192,19 @@ def sample_labels(self) -> List:
                 self._sample_labels.append(label)
         return self._sample_labels
 
+    def translate_label_id_to_name(self, class_id: int) -> str:
```
Comment (matanper) [inline deepchecks/vision/dataset.py:195]: I'd drop the word translate

File: deepchecks/vision/checks/distribution/train_test_label_drift.py [hunk 4]
```diff
@@ -218,10 +218,12 @@ def compute(self, context: Context) -> CheckResult:
...
-        train_discrete_hists = iter([{k: self._train_counters[i][k] for k in all_discrete_categories[i]} for i in
-                                     range(len(self._discrete_label_measurements))])
-        test_discrete_hists = iter([{k: self._test_counters[i][k] for k in all_discrete_categories[i]} for i in
-                                    range(len(self._discrete_label_measurements))])
+        train_discrete_hists = iter([{context.train.translate_label_id_to_name(k): self._train_counters[i][k]
+                                      for k in all_discrete_categories[i]}
+                                    for i in range(len(self._discrete_label_measurements))])
+        test_discrete_hists = iter([{context.test.translate_label_id_to_name(k): self._test_counters[i][k]
+                                     for k in all_discrete_categories[i]}
+                                    for i in range(len(self._discrete_label_measurements))])
```
[the remark this answers is not in the document]
Comment (nirhutnik) [reply deepchecks/vision/checks/distribution/train_test_label_drift.py:226]: Not sure what you mean by this.

The measurements have to be run after the conversion, as we discussed. So even if you pass dataset as a parameter, you still need to first convert and then run the measurement.

[the remark this answers is not in the document]
Comment (nirhutnik) [reply deepchecks/vision/checks/distribution/train_test_label_drift.py:226]: Regarding making the measurements work with lists - you mean to convert the tensors to lists? I have no issue with that.
Anyway, we need them to work per batch but that's for another time

**Review by matanper — APPROVED** [review]

**Review by JKL98ISR — APPROVED** [review]

**Pull request merged**